### PR TITLE
Fix for issue #5631

### PR DESCRIPTION
**Pull request by ercanse**
Fixes the flickering hovercard issue reported in https://github.com/diaspora/diaspora/issues/5631.

### CODE REVIEW

**Comment by AugierLe42e** [comment]
Thx, this one annoyed me >.<

**Comment by jhass** [comment]
Sorry, but not an improvement in all cases for me: http://cloud.aeshna.de/u/mrzyx/hovercard.webm

I think it also changes look & feel quite a bit in the other places. Isn't there a way to debounce the additional events?

**Comment by S-Ercan** [comment]
Ok, I've reverted the changes regarding hovercard positioning and used a different approach now to solve the flickering issue.

I've modified the `_mouseleaveHandler` logic so that, while `mouseleave` events can still be fired from both the parent and the hovercard, they are processed only if we have left both.
If you leave the hovercard but are still in the parent, or you leave the parent but are still in the hovercard, the handler returns false.
The absence of the first restriction was what caused the flickering problem.

File: app/assets/javascripts/app/views/hovercard_view.js [hunk 1]
```diff
@@ -140,6 +143,14 @@ app.views.Hovercard = app.views.Base.extend({
       top: p_pos.top + p_height - 25,
       left: p_pos.left
     });
-  }
+  },
+  
+  mouseIsOverElement: function(element, event) {
+  	var el_pos = element.offset();
```
Comment (jhass) [inline app/assets/javascripts/app/views/hovercard_view.js]: We use two spaces for indentation I think, @svbergerem do you know why Hound isn't complaining about this?

Comment (svbergerem) [inline app/assets/javascripts/app/views/hovercard_view.js]: Could be related to https://github.com/thoughtbot/hound/issues/418.

File: app/assets/javascripts/app/views/hovercard_view.js [hunk 2]
```diff
@@ -140,6 +143,14 @@ app.views.Hovercard = app.views.Base.extend({
...
       left: p_pos.left
     });
-  }
+  },
+  
+  mouseIsOverElement: function(element, event) {
+    var el_pos = element.offset();
+    return event.pageX >= el_pos.left && 
+      event.pageX <= el_pos.left + element.width() &&
+  	  event.pageY >= el_pos.top &&
```
Comment (jhass) [inline app/assets/javascripts/app/views/hovercard_view.js]: Please make sure to only use spaces to indent.

File: app/assets/javascripts/app/views/hovercard_view.js [hunk 3]
```diff
@@ -140,6 +143,14 @@ app.views.Hovercard = app.views.Base.extend({
...
     });
-  }
+  },
+  
+  mouseIsOverElement: function(element, event) {
+    var el_pos = element.offset();
+    return event.pageX >= el_pos.left && 
+      event.pageX <= el_pos.left + element.width() &&
+  	  event.pageY >= el_pos.top && 
+  	  event.pageY <= el_pos.top + element.height();
```
Comment (jhass) [inline app/assets/javascripts/app/views/hovercard_view.js]: This line has a hard tab too.

**Comment by jhass** [comment]
Awesome this works great. Please just make sure to resolve the above comments and squash your commits into one:

```bash
git remote add upstream git://github.com/diaspora/diaspora.git
git fetch upstream
git checkout 5631-broken-people-preview
git rebase -i upstream/develop
# Choose pick for the first line, squash for the other ones, save & quit
git push -f origin 5631-broken-people-preview
```

As a tip for the future: You can edit a commit with `git commit --amend`, which is, given you're in a feature branch, is fine to do even after you pushed it.

**Comment by S-Ercan** [comment]
Here goes! Indentation fixed, commits squashed.
Thanks for your feedback!

**Comment by jhass** [comment]
Thank you!

**Comment by Flaburgan** [comment]
Thanks!

**Pull request merged**